From 73c3239101dbc6556fb417f834ffa278832b3958 Mon Sep 17 00:00:00 2001
From: "guoyin.chen" <guoyin.chen@freescale.com>
Date: Mon, 29 Jul 2013 15:35:02 +0800
Subject: [PATCH 223/602] ENGR00272939 Update fstab.freescale as jb4.3 vold
 upgrade

New Vold in JB4.3 will use fstab.freescale to get volume config, which
is same as it in init.
Make each board config have their own fstab.freescale

Signed-off-by: guoyin.chen <guoyin.chen@freescale.com>
---
 arm2_6dq/BoardConfig.mk             | 10 ++++++++++
 arm2_6dq/fstab.freescale            | 12 ++++++++++++
 arm2_6dq/vold.fstab                 | 36 -----------------------------------
 arm2_6sl/BoardConfig.mk             | 10 ++++++++++
 arm2_6sl/fstab.freescale            | 12 ++++++++++++
 arm2_6sl/vold.fstab                 | 36 -----------------------------------
 evk_6sl/fstab.freescale             | 12 ++++++++++++
 evk_6sl/vold.fstab                  | 36 -----------------------------------
 hdmidongle_6dq/BoardConfig.mk       | 10 ++++++++++
 hdmidongle_6dq/fstab.freescale      | 12 ++++++++++++
 hdmidongle_6dq/fstab_nand.freescale | 13 +++++++++++++
 hdmidongle_6dq/vold.fstab           | 38 -------------------------------------
 imx53_smd/fstab.freescale           |  5 +++--
 imx53_smd/vold.fstab                | 34 ---------------------------------
 imx5x/imx53_smd.mk                  |  1 -
 imx6/arm2_6dq.mk                    |  4 +++-
 imx6/arm2_6sl.mk                    |  4 +++-
 imx6/etc/fstab.freescale            |  7 ++++---
 imx6/evk_6sl.mk                     |  4 +++-
 imx6/hdmidongle_6dq.mk              |  4 +++-
 imx6/imx6.mk                        |  2 --
 imx6/imx6_target_fs.mk              |  4 ----
 imx6/sabreauto_6q.mk                |  4 +++-
 imx6/sabresd_6dq.mk                 |  4 +++-
 sabreauto_6q/BoardConfig.mk         | 10 ++++++++++
 sabreauto_6q/fstab.freescale        | 12 ++++++++++++
 sabreauto_6q/fstab_nand.freescale   | 13 +++++++++++++
 sabreauto_6q/vold.fstab             | 35 ----------------------------------
 sabresd_6dq/BoardConfig.mk          | 10 ++++++++++
 sabresd_6dq/fstab.freescale         | 12 ++++++++++++
 sabresd_6dq/vold.fstab              | 38 -------------------------------------
 31 files changed, 173 insertions(+), 271 deletions(-)
 create mode 100644 arm2_6dq/fstab.freescale
 delete mode 100644 arm2_6dq/vold.fstab
 create mode 100644 arm2_6sl/fstab.freescale
 delete mode 100644 arm2_6sl/vold.fstab
 create mode 100644 evk_6sl/fstab.freescale
 delete mode 100644 evk_6sl/vold.fstab
 create mode 100644 hdmidongle_6dq/fstab.freescale
 create mode 100644 hdmidongle_6dq/fstab_nand.freescale
 delete mode 100644 hdmidongle_6dq/vold.fstab
 delete mode 100644 imx53_smd/vold.fstab
 create mode 100644 sabreauto_6q/fstab.freescale
 create mode 100644 sabreauto_6q/fstab_nand.freescale
 delete mode 100644 sabreauto_6q/vold.fstab
 create mode 100644 sabresd_6dq/fstab.freescale
 delete mode 100644 sabresd_6dq/vold.fstab

diff --git a/arm2_6dq/BoardConfig.mk b/arm2_6dq/BoardConfig.mk
index 1e2ac24..12f8b54 100755
--- a/arm2_6dq/BoardConfig.mk
+++ b/arm2_6dq/BoardConfig.mk
@@ -11,6 +11,16 @@ include device/fsl/imx6/BoardConfigCommon.mk
 BUILD_TARGET_FS ?= ext4
 include device/fsl/imx6/imx6_target_fs.mk
 
+ifeq ($(BUILD_TARGET_FS),ubifs)
+# build ubifs for nand devices
+PRODUCT_COPY_FILES +=	\
+	device/fsl/arm2_6dq/fstab_nand.freescale:root/fstab.freescale
+else
+# build for ext4
+PRODUCT_COPY_FILES +=	\
+	device/fsl/arm2_6dq/fstab.freescale:root/fstab.freescale
+endif # BUILD_TARGET_FS
+
 TARGET_BOOTLOADER_BOARD_NAME := ARM2
 
 BOARD_SOC_CLASS := IMX6
diff --git a/arm2_6dq/fstab.freescale b/arm2_6dq/fstab.freescale
new file mode 100644
index 0000000..b81ba34
--- /dev/null
+++ b/arm2_6dq/fstab.freescale
@@ -0,0 +1,12 @@
+# Android fstab file.
+#<src>                                                  <mnt_point>         <type>    <mnt_flags and options>                       <fs_mgr_flags>
+# The filesystem that contains the filesystem checker binary (typically /system) cannot
+# specify MF_CHECK, and must come before any filesystems that do specify MF_CHECK
+
+/devices/platform/sdhci-esdhc-imx.2/mmc_host/mmc2 /mnt/extsd vfat defaults voldmanaged=sdcard:auto
+/devices/platform/fsl-ehci /mnt/udisk vfat defaults voldmanaged=sdcard:auto
+/dev/block/mmcblk0p5    /system  ext4    ro                                                                               wait
+/dev/block/mmcblk0p4    /data    ext4    nosuid,nodev,nodiratime,noatime,nomblk_io_submit,noauto_da_alloc,errors=panic    wait,encryptable=footer
+/dev/block/mmcblk0p6    /cache   ext4    nosuid,nodev,nomblk_io_submit				                          wait
+/dev/block/mmcblk0p7    /device  ext4    ro,nosuid,nodev				                                  wait
+
diff --git a/arm2_6dq/vold.fstab b/arm2_6dq/vold.fstab
deleted file mode 100644
index 2344535..0000000
--- a/arm2_6dq/vold.fstab
+++ /dev/null
@@ -1,36 +0,0 @@
-## Vold 2.0 Generic fstab
-## - San Mehat (san@android.com)
-##
-
-#######################
-## Regular device mount
-##
-## Format: dev_mount <label> <mount_point> <part> <sysfs_path1...>
-## label        - Label for the volume
-## mount_point  - Where the volume will be mounted
-## part         - Partition # (1 based), or 'auto' for first usable partition.
-## <sysfs_path> - List of sysfs paths to source devices
-######################
-
-## Example of a standard sdcard mount for the emulator / Dream
-# Mounts the first usable partition of the specified device
-#for internal emmc as primary storage
-#dev_mount sdcard /mnt/sdcard auto /devices/platform/mxsdhci.2/mmc_host/mmc0
-#for external sdcard as primary storage
-#dev_mount sdcard /mnt/sdcard 4 /devices/platform/sdhci-esdhc-imx.0/mmc_host/mmc0/
-dev_mount extsd  /mnt/extsd auto /devices/platform/sdhci-esdhc-imx.2/mmc_host/mmc2/ /devices/platform/sdhci-esdhc-imx.2/mmc_host/mmc1/
-#dev_mount extsd /mnt/extsd auto /devices/platform/mxsdhci.0/mmc_host/mmc1
-#used for usb host
-#dev_mount udisk /mnt/udisk auto /devices/platform/fsl-ehci.0/usb1
-#used for usb otg host
-#dev_mount udisk /mnt/udisk auto /devices/platform/fsl-ehci.0/usb
-#used for all usb host
-dev_mount udisk /mnt/udisk auto /devices/platform/fsl-ehci
-
-
-## Example of a dual card setup
-# dev_mount left_sdcard  /sdcard1  auto /devices/platform/goldfish_mmc.0 /devices/platform/msm_sdcc.2/mmc_host/mmc1
-# dev_mount right_sdcard /sdcard2  auto /devices/platform/goldfish_mmc.1 /devices/platform/msm_sdcc.3/mmc_host/mmc1
-
-## Example of specifying a specific partition for mounts
-# dev_mount sdcard /sdcard 2 /devices/platform/goldfish_mmc.0 /devices/platform/msm_sdcc.2/mmc_host/mmc1
diff --git a/arm2_6sl/BoardConfig.mk b/arm2_6sl/BoardConfig.mk
index d12f9e7..5029c9d 100755
--- a/arm2_6sl/BoardConfig.mk
+++ b/arm2_6sl/BoardConfig.mk
@@ -11,6 +11,16 @@ include device/fsl/imx6/BoardConfigCommon.mk
 BUILD_TARGET_FS ?= ext4
 include device/fsl/imx6/imx6_target_fs.mk
 
+ifeq ($(BUILD_TARGET_FS),ubifs)
+# build ubifs for nand devices
+PRODUCT_COPY_FILES +=	\
+	device/fsl/arm2_6sl/fstab_nand.freescale:root/fstab.freescale
+else
+# build for ext4
+PRODUCT_COPY_FILES +=	\
+	device/fsl/arm2_6sl/fstab.freescale:root/fstab.freescale
+endif # BUILD_TARGET_FS
+
 TARGET_BOOTLOADER_BOARD_NAME := ARM2
 
 BOARD_SOC_CLASS := IMX6
diff --git a/arm2_6sl/fstab.freescale b/arm2_6sl/fstab.freescale
new file mode 100644
index 0000000..dcbcd14
--- /dev/null
+++ b/arm2_6sl/fstab.freescale
@@ -0,0 +1,12 @@
+# Android fstab file.
+#<src>                                                  <mnt_point>         <type>    <mnt_flags and options>                       <fs_mgr_flags>
+# The filesystem that contains the filesystem checker binary (typically /system) cannot
+# specify MF_CHECK, and must come before any filesystems that do specify MF_CHECK
+
+/devices/platform/sdhci-esdhc-imx.2/mmc_host/mmc1 /mnt/extsd vfat defaults voldmanaged=sdcard:auto
+/devices/platform/fsl-ehci /mnt/udisk vfat defaults voldmanaged=sdcard:auto
+/dev/block/mmcblk0p5    /system  ext4    ro                                                                               wait
+/dev/block/mmcblk0p4    /data    ext4    nosuid,nodev,nodiratime,noatime,nomblk_io_submit,noauto_da_alloc,errors=panic    wait,encryptable=footer
+/dev/block/mmcblk0p6    /cache   ext4    nosuid,nodev,nomblk_io_submit				                          wait
+/dev/block/mmcblk0p7    /device  ext4    ro,nosuid,nodev				                                  wait
+
diff --git a/arm2_6sl/vold.fstab b/arm2_6sl/vold.fstab
deleted file mode 100644
index 2344535..0000000
--- a/arm2_6sl/vold.fstab
+++ /dev/null
@@ -1,36 +0,0 @@
-## Vold 2.0 Generic fstab
-## - San Mehat (san@android.com)
-##
-
-#######################
-## Regular device mount
-##
-## Format: dev_mount <label> <mount_point> <part> <sysfs_path1...>
-## label        - Label for the volume
-## mount_point  - Where the volume will be mounted
-## part         - Partition # (1 based), or 'auto' for first usable partition.
-## <sysfs_path> - List of sysfs paths to source devices
-######################
-
-## Example of a standard sdcard mount for the emulator / Dream
-# Mounts the first usable partition of the specified device
-#for internal emmc as primary storage
-#dev_mount sdcard /mnt/sdcard auto /devices/platform/mxsdhci.2/mmc_host/mmc0
-#for external sdcard as primary storage
-#dev_mount sdcard /mnt/sdcard 4 /devices/platform/sdhci-esdhc-imx.0/mmc_host/mmc0/
-dev_mount extsd  /mnt/extsd auto /devices/platform/sdhci-esdhc-imx.2/mmc_host/mmc2/ /devices/platform/sdhci-esdhc-imx.2/mmc_host/mmc1/
-#dev_mount extsd /mnt/extsd auto /devices/platform/mxsdhci.0/mmc_host/mmc1
-#used for usb host
-#dev_mount udisk /mnt/udisk auto /devices/platform/fsl-ehci.0/usb1
-#used for usb otg host
-#dev_mount udisk /mnt/udisk auto /devices/platform/fsl-ehci.0/usb
-#used for all usb host
-dev_mount udisk /mnt/udisk auto /devices/platform/fsl-ehci
-
-
-## Example of a dual card setup
-# dev_mount left_sdcard  /sdcard1  auto /devices/platform/goldfish_mmc.0 /devices/platform/msm_sdcc.2/mmc_host/mmc1
-# dev_mount right_sdcard /sdcard2  auto /devices/platform/goldfish_mmc.1 /devices/platform/msm_sdcc.3/mmc_host/mmc1
-
-## Example of specifying a specific partition for mounts
-# dev_mount sdcard /sdcard 2 /devices/platform/goldfish_mmc.0 /devices/platform/msm_sdcc.2/mmc_host/mmc1
diff --git a/evk_6sl/fstab.freescale b/evk_6sl/fstab.freescale
new file mode 100644
index 0000000..0d20f60
--- /dev/null
+++ b/evk_6sl/fstab.freescale
@@ -0,0 +1,12 @@
+# Android fstab file.
+#<src>                                                  <mnt_point>         <type>    <mnt_flags and options>                       <fs_mgr_flags>
+# The filesystem that contains the filesystem checker binary (typically /system) cannot
+# specify MF_CHECK, and must come before any filesystems that do specify MF_CHECK
+
+/devices/platform/sdhci-esdhc-imx.0/mmc_host/mmc1 /mnt/extsd vfat defaults voldmanaged=sdcard:auto
+/devices/platform/fsl-ehci /mnt/udisk vfat defaults voldmanaged=sdcard:auto
+/dev/block/mmcblk0p5    /system  ext4    ro                                                                               wait
+/dev/block/mmcblk0p4    /data    ext4    nosuid,nodev,nodiratime,noatime,nomblk_io_submit,noauto_da_alloc,errors=panic    wait,encryptable=footer
+/dev/block/mmcblk0p6    /cache   ext4    nosuid,nodev,nomblk_io_submit				                          wait
+/dev/block/mmcblk0p7    /device  ext4    ro,nosuid,nodev				                                  wait
+
diff --git a/evk_6sl/vold.fstab b/evk_6sl/vold.fstab
deleted file mode 100644
index a89293a..0000000
--- a/evk_6sl/vold.fstab
+++ /dev/null
@@ -1,36 +0,0 @@
-## Vold 2.0 Generic fstab
-## - San Mehat (san@android.com)
-##
-
-#######################
-## Regular device mount
-##
-## Format: dev_mount <label> <mount_point> <part> <sysfs_path1...>
-## label        - Label for the volume
-## mount_point  - Where the volume will be mounted
-## part         - Partition # (1 based), or 'auto' for first usable partition.
-## <sysfs_path> - List of sysfs paths to source devices
-######################
-
-## Example of a standard sdcard mount for the emulator / Dream
-# Mounts the first usable partition of the specified device
-#for internal emmc as primary storage
-#dev_mount sdcard /mnt/sdcard auto /devices/platform/mxsdhci.2/mmc_host/mmc0
-#for external sdcard as primary storage
-#dev_mount sdcard /mnt/sdcard 4 /devices/platform/sdhci-esdhc-imx.1/mmc_host/mmc0/
-dev_mount extsd  /mnt/extsd auto /devices/platform/sdhci-esdhc-imx.0/mmc_host/mmc1/ /devices/platform/sdhci-esdhc-imx.2/mmc_host/mmc2/
-#dev_mount extsd /mnt/extsd auto /devices/platform/mxsdhci.0/mmc_host/mmc1
-#used for usb host
-#dev_mount udisk /mnt/udisk auto /devices/platform/fsl-ehci.0/usb1
-#used for usb otg host
-#dev_mount udisk /mnt/udisk auto /devices/platform/fsl-ehci.0/usb
-#used for all usb host
-dev_mount udisk /mnt/udisk auto /devices/platform/fsl-ehci
-
-
-## Example of a dual card setup
-# dev_mount left_sdcard  /sdcard1  auto /devices/platform/goldfish_mmc.0 /devices/platform/msm_sdcc.2/mmc_host/mmc1
-# dev_mount right_sdcard /sdcard2  auto /devices/platform/goldfish_mmc.1 /devices/platform/msm_sdcc.3/mmc_host/mmc1
-
-## Example of specifying a specific partition for mounts
-# dev_mount sdcard /sdcard 2 /devices/platform/goldfish_mmc.0 /devices/platform/msm_sdcc.2/mmc_host/mmc1
diff --git a/hdmidongle_6dq/BoardConfig.mk b/hdmidongle_6dq/BoardConfig.mk
index c3736dd..12aa8d9 100755
--- a/hdmidongle_6dq/BoardConfig.mk
+++ b/hdmidongle_6dq/BoardConfig.mk
@@ -10,6 +10,16 @@ include device/fsl-proprietary/gpu-viv/fsl-gpu.mk
 BUILD_TARGET_FS ?= ubifs
 include device/fsl/imx6/imx6_target_fs.mk
 
+ifeq ($(BUILD_TARGET_FS),ubifs)
+# build ubifs for nand devices
+PRODUCT_COPY_FILES +=	\
+	device/fsl/hdmidongle_6dq/fstab_nand.freescale:root/fstab.freescale
+else
+# build for ext4
+PRODUCT_COPY_FILES +=	\
+	device/fsl/hdmidongle_6dq/fstab.freescale:root/fstab.freescale
+endif # BUILD_TARGET_FS
+
 TARGET_BOOTLOADER_BOARD_NAME := HDMIDONGLE
 PRODUCT_MODEL := HDMIDONGLE-MX6DQ
 
diff --git a/hdmidongle_6dq/fstab.freescale b/hdmidongle_6dq/fstab.freescale
new file mode 100644
index 0000000..65e1db6
--- /dev/null
+++ b/hdmidongle_6dq/fstab.freescale
@@ -0,0 +1,12 @@
+# Android fstab file.
+#<src>                                                  <mnt_point>         <type>    <mnt_flags and options>                       <fs_mgr_flags>
+# The filesystem that contains the filesystem checker binary (typically /system) cannot
+# specify MF_CHECK, and must come before any filesystems that do specify MF_CHECK
+
+/devices/platform/sdhci-esdhc-imx.1/mmc_host/mmc1 /mnt/extsd vfat defaults voldmanaged=sdcard:auto
+/devices/platform/fsl-ehci /mnt/udisk vfat defaults voldmanaged=sdcard:auto
+/dev/block/mmcblk0p5    /system  ext4    ro                                                                               wait
+/dev/block/mmcblk0p4    /data    ext4    nosuid,nodev,nodiratime,noatime,nomblk_io_submit,noauto_da_alloc,errors=panic    wait,encryptable=footer
+/dev/block/mmcblk0p6    /cache   ext4    nosuid,nodev,nomblk_io_submit				                          wait
+/dev/block/mmcblk0p7    /device  ext4    ro,nosuid,nodev				                                  wait
+
diff --git a/hdmidongle_6dq/fstab_nand.freescale b/hdmidongle_6dq/fstab_nand.freescale
new file mode 100644
index 0000000..99a0ed4
--- /dev/null
+++ b/hdmidongle_6dq/fstab_nand.freescale
@@ -0,0 +1,13 @@
+# Android fstab file.
+#<src>                                                  <mnt_point>         <type>    <mnt_flags>                                                                         <fs_mgr_flags>
+# The filesystem that contains the filesystem checker binary (typically /system) cannot
+# specify MF_CHECK, and must come before any filesystems that do specify MF_CHECK
+
+# below is for ubi support, uncomment to support ubifs(NAND)
+
+/devices/platform/sdhci-esdhc-imx.1/mmc_host/mmc1 /mnt/extsd vfat defaults voldmanaged=sdcard:auto
+/devices/platform/fsl-ehci /mnt/udisk vfat defaults voldmanaged=sdcard:auto
+ubi0:system	/system		ubifs 	ro					wait
+ubi0:data       /data           ubifs   nosuid,nodev,nodiratime,noatime         wait
+ubi0:cache      /cache          ubifs   nosuid,nodev                            wait
+ubi0:device     /device         ubifs   nosuid,nodev                         wait
diff --git a/hdmidongle_6dq/vold.fstab b/hdmidongle_6dq/vold.fstab
deleted file mode 100644
index dffc61e..0000000
--- a/hdmidongle_6dq/vold.fstab
+++ /dev/null
@@ -1,38 +0,0 @@
-## Vold 2.0 Generic fstab
-## - San Mehat (san@android.com)
-##
-
-#######################
-## Regular device mount
-##
-## Format: dev_mount <label> <mount_point> <part> <sysfs_path1...>
-## label        - Label for the volume
-## mount_point  - Where the volume will be mounted
-## part         - Partition # (1 based), or 'auto' for first usable partition.
-## <sysfs_path> - List of sysfs paths to source devices
-######################
-
-## Example of a standard sdcard mount for the emulator / Dream
-# Mounts the first usable partition of the specified device
-#for internal emmc as primary storage
-#dev_mount sdcard /mnt/sdcard auto /devices/platform/mxsdhci.2/mmc_host/mmc0
-#for external sdcard as primary storage
-#dev_mount sdcard /mnt/sdcard auto /devices/platform/sdhci-esdhc-imx.3/mmc_host/mmc0/
-#dev_mount extsd /mnt/extsd auto /devices/platform/mxsdhci.0/mmc_host/mmc1
-#used for usb host
-#dev_mount udisk /mnt/udisk auto /devices/platform/fsl-ehci.0/usb1
-#used for usb otg host
-#dev_mount udisk /mnt/udisk auto /devices/platform/fsl-ehci.0/usb
-#used for all usb host
-dev_mount udisk /mnt/udisk auto /devices/platform/fsl-ehci
-#mount SDHC4 SD card /mnt/sdcard as primary storage forMX6Q SABER_LITE RevC
-dev_mount extsd /mnt/extsd auto /devices/platform/sdhci-esdhc-imx.1/mmc_host/mmc1
-#mount SDHC3 TF card to /mnt/extsd as external storage forMX6Q SABER_LITE RevC
-#dev_mount sdcard /mnt/sdcard 1 /devices/platform/sdhci-esdhc-imx.2/mmc_host/mmc0
-
-## Example of a dual card setup
-# dev_mount left_sdcard  /sdcard1  auto /devices/platform/goldfish_mmc.0 /devices/platform/msm_sdcc.2/mmc_host/mmc1
-# dev_mount right_sdcard /sdcard2  auto /devices/platform/goldfish_mmc.1 /devices/platform/msm_sdcc.3/mmc_host/mmc1
-
-## Example of specifying a specific partition for mounts
-# dev_mount sdcard /sdcard 2 /devices/platform/goldfish_mmc.0 /devices/platform/msm_sdcc.2/mmc_host/mmc1
diff --git a/imx53_smd/fstab.freescale b/imx53_smd/fstab.freescale
index 92b6df2..de51709 100644
--- a/imx53_smd/fstab.freescale
+++ b/imx53_smd/fstab.freescale
@@ -1,9 +1,10 @@
 # Android fstab file.
-#<src>                                                  <mnt_point>         <type>    <mnt_flags>                                                                         <fs_mgr_flags>
+#<src>                                                  <mnt_point>         <type>    <mnt_flags and options>                       <fs_mgr_flags>
 # The filesystem that contains the filesystem checker binary (typically /system) cannot
 # specify MF_CHECK, and must come before any filesystems that do specify MF_CHECK
 
-
+/devices/platform/sdhci-esdhc-imx.0/mmc_host/mmc2 /mnt/extsd vfat defaults voldmanaged=sdcard:auto
+/devices/platform/fsl-ehci /mnt/udisk vfat defaults voldmanaged=sdcard:auto
 /dev/block/mmcblk0p2           /system             ext4      ro                                                    wait
 /dev/block/mmcblk0p6           /cache              ext4      noatime,nosuid,nodev,nomblk_io_submit,errors=panic    wait
 /dev/block/mmcblk0p5           /data               ext4      noatime,nosuid,nodev,nomblk_io_submit,errors=panic    wait
diff --git a/imx53_smd/vold.fstab b/imx53_smd/vold.fstab
deleted file mode 100644
index 561fb87..0000000
--- a/imx53_smd/vold.fstab
+++ /dev/null
@@ -1,34 +0,0 @@
-## Vold 2.0 Generic fstab
-## - San Mehat (san@android.com)
-##
-
-#######################
-## Regular device mount
-##
-## Format: dev_mount <label> <mount_point> <part> <sysfs_path1...>
-## label        - Label for the volume
-## mount_point  - Where the volume will be mounted
-## part         - Partition # (1 based), or 'auto' for first usable partition.
-## <sysfs_path> - List of sysfs paths to source devices
-######################
-
-## Example of a standard sdcard mount for the emulator / Dream
-# Mounts the first usable partition of the specified device
-#for internal emmc as primary storage
-#dev_mount sdcard /mnt/sdcard 1 /devices/platform/sdhci-esdhc-imx.0/mmc_host/mmc0 /devices/platform/sdhci-esdhc-imx.2/mmc_host/mmc0
-#for external sdcard as primary storage
-dev_mount extsd /mnt/extsd auto /devices/platform/sdhci-esdhc-imx.2/mmc_host/mmc2 /devices/platform/sdhci-esdhc-imx.0/mmc_host/mmc2
-#used for usb host
-#dev_mount udisk /mnt/udisk auto /devices/platform/fsl-ehci.1/usb /devices/platform/fsl-ehci.0/usb
-#used for usb otg host
-#dev_mount udisk /mnt/udisk auto /devices/platform/fsl-ehci.0/usb
-#used for all usb host
-dev_mount udisk /mnt/udisk auto /devices/platform/fsl-ehci
-
-
-## Example of a dual card setup
-# dev_mount left_sdcard  /sdcard1  auto /devices/platform/goldfish_mmc.0 /devices/platform/msm_sdcc.2/mmc_host/mmc1
-# dev_mount right_sdcard /sdcard2  auto /devices/platform/goldfish_mmc.1 /devices/platform/msm_sdcc.3/mmc_host/mmc1
-
-## Example of specifying a specific partition for mounts
-# dev_mount sdcard /sdcard 2 /devices/platform/goldfish_mmc.0 /devices/platform/msm_sdcc.2/mmc_host/mmc1
diff --git a/imx5x/imx53_smd.mk b/imx5x/imx53_smd.mk
index 1446411..ec1efe6 100644
--- a/imx5x/imx53_smd.mk
+++ b/imx5x/imx53_smd.mk
@@ -11,7 +11,6 @@ PRODUCT_COPY_FILES += \
 	device/common/gps/gps.conf_US:system/etc/gps.conf \
 	device/fsl/imx53_smd/fstab.freescale:root/fstab.freescale \
 	device/fsl/imx53_smd/init.freescale.rc:root/init.freescale.rc \
-        device/fsl/imx53_smd/vold.fstab:system/etc/vold.fstab \
 	device/fsl/imx53_smd/gpsreset.sh:system/bin/gpsreset.sh \
 	device/fsl/common/input/eGalax_Touch_Screen.idc:system/usr/idc/eGalax_Touch_Screen.idc \
 	device/fsl/common/input/eGalax_Touch_Screen.idc:system/usr/idc/HannStar_P1003_Touchscreen.idc 
diff --git a/imx6/arm2_6dq.mk b/imx6/arm2_6dq.mk
index c649e96..ffc3cf6 100644
--- a/imx6/arm2_6dq.mk
+++ b/imx6/arm2_6dq.mk
@@ -3,13 +3,15 @@
 
 $(call inherit-product, device/fsl/imx6/imx6.mk)
 
+$(shell touch device/fsl/arm2_6dq/fstab_nand.freescale)
+$(shell touch device/fsl/arm2_6dq/fstab.freescale)
+
 # Overrides
 PRODUCT_NAME := arm2_6dq
 PRODUCT_DEVICE := arm2_6dq
 
 PRODUCT_COPY_FILES += \
 	device/fsl/arm2_6dq/required_hardware.xml:system/etc/permissions/required_hardware.xml \
-	device/fsl/arm2_6dq/vold.fstab:system/etc/vold.fstab \
 	device/fsl/arm2_6dq/init.rc:root/init.freescale.rc
 
 DEVICE_PACKAGE_OVERLAYS := device/fsl/arm2_6dq/overlay
diff --git a/imx6/arm2_6sl.mk b/imx6/arm2_6sl.mk
index 2b9d943..08f75b4 100644
--- a/imx6/arm2_6sl.mk
+++ b/imx6/arm2_6sl.mk
@@ -3,13 +3,15 @@
 
 $(call inherit-product, device/fsl/imx6/imx6.mk)
 
+$(shell touch device/fsl/arm2_6sl/fstab_nand.freescale)
+$(shell touch device/fsl/arm2_6sl/fstab.freescale)
+
 # Overrides
 PRODUCT_NAME := arm2_6sl
 PRODUCT_DEVICE := arm2_6sl
 
 PRODUCT_COPY_FILES += \
 	device/fsl/arm2_6sl/required_hardware.xml:system/etc/permissions/required_hardware.xml \
-	device/fsl/arm2_6sl/vold.fstab:system/etc/vold.fstab \
 	device/fsl/arm2_6sl/init.rc:root/init.freescale.rc \
 	device/fsl/arm2_6sl/mxckpd.kl:system/usr/keylayout/mxckpd.kl
 
diff --git a/imx6/etc/fstab.freescale b/imx6/etc/fstab.freescale
index 004bc46..e5001be 100644
--- a/imx6/etc/fstab.freescale
+++ b/imx6/etc/fstab.freescale
@@ -1,9 +1,10 @@
+
 # Android fstab file.
-#<src>                                                  <mnt_point>         <type>    <mnt_flags>                                                                         <fs_mgr_flags>
+#<src>                                                  <mnt_point>         <type>    <mnt_flags and options>                       <fs_mgr_flags>
 # The filesystem that contains the filesystem checker binary (typically /system) cannot
 # specify MF_CHECK, and must come before any filesystems that do specify MF_CHECK
-
-
+/devices/platform/sdhci-esdhc-imx.2/mmc_host/mmc1 /mnt/extsd vfat defaults voldmanaged=sdcard:auto
+/devices/platform/fsl-ehci /mnt/udisk vfat defaults voldmanaged=sdcard:auto
 /dev/block/mmcblk0p5    /system  ext4    ro                                                                               wait
 /dev/block/mmcblk0p4    /data    ext4    nosuid,nodev,nodiratime,noatime,nomblk_io_submit,noauto_da_alloc,errors=panic    wait,encryptable=footer
 /dev/block/mmcblk0p6    /cache   ext4    nosuid,nodev,nomblk_io_submit				                          wait
diff --git a/imx6/evk_6sl.mk b/imx6/evk_6sl.mk
index a50c8bb..dae9c6a 100644
--- a/imx6/evk_6sl.mk
+++ b/imx6/evk_6sl.mk
@@ -4,13 +4,15 @@
 $(call inherit-product, device/fsl/imx6/imx6.mk)
 $(call inherit-product-if-exists,vendor/google/products/gms.mk)
 
+$(shell touch device/fsl/evk_6sl/fstab_nand.freescale)
+$(shell touch device/fsl/evk_6sl/fstab.freescale)
+
 # Overrides
 PRODUCT_NAME := evk_6sl
 PRODUCT_DEVICE := evk_6sl
 
 PRODUCT_COPY_FILES += \
 	device/fsl/evk_6sl/required_hardware.xml:system/etc/permissions/required_hardware.xml \
-	device/fsl/evk_6sl/vold.fstab:system/etc/vold.fstab \
 	device/fsl/evk_6sl/init.rc:root/init.freescale.rc \
 	device/fsl/common/input/imx-keypad.idc:system/usr/idc/imx-keypad.idc \
 	device/fsl/common/input/imx-keypad.kl:system/usr/keylayout/imx-keypad.kl \
diff --git a/imx6/hdmidongle_6dq.mk b/imx6/hdmidongle_6dq.mk
index b6c13c4..53523c0 100644
--- a/imx6/hdmidongle_6dq.mk
+++ b/imx6/hdmidongle_6dq.mk
@@ -4,6 +4,9 @@
 $(call inherit-product, device/fsl/imx6/imx6.mk)
 $(call inherit-product-if-exists,vendor/google/products/gms.mk)
 
+$(shell touch device/fsl/hdmidongle_6dq/fstab_nand.freescale)
+$(shell touch device/fsl/hdmidongle_6dq/fstab.freescale)
+
 # Overrides
 PRODUCT_NAME := hdmidongle_6dq
 PRODUCT_DEVICE := hdmidongle_6dq
@@ -11,7 +14,6 @@ PRODUCT_DEVICE := hdmidongle_6dq
 PRODUCT_COPY_FILES += \
 	device/fsl/hdmidongle_6dq/required_hardware.xml:system/etc/permissions/required_hardware.xml \
 	device/fsl/hdmidongle_6dq/init.rc:root/init.freescale.rc \
-	device/fsl/hdmidongle_6dq/vold.fstab:system/etc/vold.fstab \
 	device/fsl/hdmidongle_6dq/audio_policy.conf:system/etc/audio_policy.conf \
 	device/fsl/hdmidongle_6dq/audio_effects.conf:system/vendor/etc/audio_effects.conf
 
diff --git a/imx6/imx6.mk b/imx6/imx6.mk
index 569f8fb..8ac0a92 100644
--- a/imx6/imx6.mk
+++ b/imx6/imx6.mk
@@ -1,8 +1,6 @@
 $(call inherit-product, $(SRC_TARGET_DIR)/product/languages_full.mk)
 $(call inherit-product, $(SRC_TARGET_DIR)/product/generic.mk)
 $(call inherit-product, $(TOPDIR)frameworks/base/data/sounds/AllAudio.mk)
-$(shell touch device/fsl/imx6/etc/fstab_nand.freescale)
-$(shell touch device/fsl/imx6/etc/fstab.freescale)
 # overrides
 PRODUCT_BRAND := Freescale
 PRODUCT_MANUFACTURER := Freescale
diff --git a/imx6/imx6_target_fs.mk b/imx6/imx6_target_fs.mk
index 8d246a3..13018ce 100644
--- a/imx6/imx6_target_fs.mk
+++ b/imx6/imx6_target_fs.mk
@@ -7,13 +7,9 @@ ifeq ($(BUILD_TARGET_FS),ubifs)
 # build ubifs for nand devices
 TARGET_USERIMAGES_USE_UBIFS := true
 TARGET_USERIMAGES_USE_EXT4 := false
-PRODUCT_COPY_FILES +=	\
-	device/fsl/imx6/etc/fstab_nand.freescale:root/fstab.freescale
 else
 # build for ext4
 TARGET_USERIMAGES_USE_EXT4 := true
 TARGET_USERIMAGES_USE_UBIFS := false
-PRODUCT_COPY_FILES +=	\
-	device/fsl/imx6/etc/fstab.freescale:root/fstab.freescale
 endif # BUILD_TARGET_FS
 
diff --git a/imx6/sabreauto_6q.mk b/imx6/sabreauto_6q.mk
index 156d977..0c04a47 100644
--- a/imx6/sabreauto_6q.mk
+++ b/imx6/sabreauto_6q.mk
@@ -3,6 +3,9 @@
 
 $(call inherit-product, device/fsl/imx6/imx6.mk)
 
+$(shell touch device/fsl/sabreauto_6q/fstab_nand.freescale)
+$(shell touch device/fsl/sabreauto_6q/fstab.freescale)
+
 # Overrides
 PRODUCT_NAME := sabreauto_6q
 PRODUCT_DEVICE := sabreauto_6q
@@ -11,7 +14,6 @@ PRODUCT_DEVICE := sabreauto_6q
 PRODUCT_COPY_FILES += \
 	device/fsl/sabreauto_6q/required_hardware.xml:system/etc/permissions/required_hardware.xml \
 	device/fsl/sabreauto_6q/init.rc:root/init.freescale.rc \
-    device/fsl/sabreauto_6q/vold.fstab:system/etc/vold.fstab \
 	device/fsl/sabreauto_6q/audio_policy.conf:system/etc/audio_policy.conf \
 	device/fsl/sabreauto_6q/audio_effects.conf:system/vendor/etc/audio_effects.conf
 
diff --git a/imx6/sabresd_6dq.mk b/imx6/sabresd_6dq.mk
index 064e9eb..028acfc 100644
--- a/imx6/sabresd_6dq.mk
+++ b/imx6/sabresd_6dq.mk
@@ -4,6 +4,9 @@
 $(call inherit-product, device/fsl/imx6/imx6.mk)
 $(call inherit-product-if-exists,vendor/google/products/gms.mk)
 
+$(shell touch device/fsl/sabresd_6dq/fstab_nand.freescale)
+$(shell touch device/fsl/sabresd_6dq/fstab.freescale)
+
 # Overrides
 PRODUCT_NAME := sabresd_6dq
 PRODUCT_DEVICE := sabresd_6dq
@@ -11,7 +14,6 @@ PRODUCT_DEVICE := sabresd_6dq
 PRODUCT_COPY_FILES += \
 	device/fsl/sabresd_6dq/required_hardware.xml:system/etc/permissions/required_hardware.xml \
 	device/fsl/sabresd_6dq/init.rc:root/init.freescale.rc \
-	device/fsl/sabresd_6dq/vold.fstab:system/etc/vold.fstab \
 	device/fsl/sabresd_6dq/gpsreset.sh:system/etc/gpsreset.sh \
 	device/fsl/sabresd_6dq/audio_policy.conf:system/etc/audio_policy.conf \
 	device/fsl/sabresd_6dq/audio_effects.conf:system/vendor/etc/audio_effects.conf
diff --git a/sabreauto_6q/BoardConfig.mk b/sabreauto_6q/BoardConfig.mk
index f7d43b5..c6505b5 100644
--- a/sabreauto_6q/BoardConfig.mk
+++ b/sabreauto_6q/BoardConfig.mk
@@ -10,6 +10,16 @@ include device/fsl-proprietary/gpu-viv/fsl-gpu.mk
 BUILD_TARGET_FS ?= ext4
 include device/fsl/imx6/imx6_target_fs.mk
 
+ifeq ($(BUILD_TARGET_FS),ubifs)
+# build ubifs for nand devices
+PRODUCT_COPY_FILES +=	\
+	device/fsl/sabreauto_6q/fstab_nand.freescale:root/fstab.freescale
+else
+# build for ext4
+PRODUCT_COPY_FILES +=	\
+	device/fsl/sabreauto_6q/fstab.freescale:root/fstab.freescale
+endif # BUILD_TARGET_FS
+
 TARGET_BOOTLOADER_BOARD_NAME := SABREAUTO
 
 BOARD_SOC_CLASS := IMX6
diff --git a/sabreauto_6q/fstab.freescale b/sabreauto_6q/fstab.freescale
new file mode 100644
index 0000000..0d20f60
--- /dev/null
+++ b/sabreauto_6q/fstab.freescale
@@ -0,0 +1,12 @@
+# Android fstab file.
+#<src>                                                  <mnt_point>         <type>    <mnt_flags and options>                       <fs_mgr_flags>
+# The filesystem that contains the filesystem checker binary (typically /system) cannot
+# specify MF_CHECK, and must come before any filesystems that do specify MF_CHECK
+
+/devices/platform/sdhci-esdhc-imx.0/mmc_host/mmc1 /mnt/extsd vfat defaults voldmanaged=sdcard:auto
+/devices/platform/fsl-ehci /mnt/udisk vfat defaults voldmanaged=sdcard:auto
+/dev/block/mmcblk0p5    /system  ext4    ro                                                                               wait
+/dev/block/mmcblk0p4    /data    ext4    nosuid,nodev,nodiratime,noatime,nomblk_io_submit,noauto_da_alloc,errors=panic    wait,encryptable=footer
+/dev/block/mmcblk0p6    /cache   ext4    nosuid,nodev,nomblk_io_submit				                          wait
+/dev/block/mmcblk0p7    /device  ext4    ro,nosuid,nodev				                                  wait
+
diff --git a/sabreauto_6q/fstab_nand.freescale b/sabreauto_6q/fstab_nand.freescale
new file mode 100644
index 0000000..3d928bb
--- /dev/null
+++ b/sabreauto_6q/fstab_nand.freescale
@@ -0,0 +1,13 @@
+# Android fstab file.
+#<src>                                                  <mnt_point>         <type>    <mnt_flags>                                                                         <fs_mgr_flags>
+# The filesystem that contains the filesystem checker binary (typically /system) cannot
+# specify MF_CHECK, and must come before any filesystems that do specify MF_CHECK
+
+# below is for ubi support, uncomment to support ubifs(NAND)
+
+/devices/platform/sdhci-esdhc-imx.0/mmc_host/mmc1 /mnt/extsd vfat defaults voldmanaged=sdcard:auto
+/devices/platform/fsl-ehci /mnt/udisk vfat defaults voldmanaged=sdcard:auto
+ubi0:system	/system		ubifs 	ro					wait
+ubi0:data       /data           ubifs   nosuid,nodev,nodiratime,noatime         wait
+ubi0:cache      /cache          ubifs   nosuid,nodev                            wait
+ubi0:device     /device         ubifs   nosuid,nodev                         wait
diff --git a/sabreauto_6q/vold.fstab b/sabreauto_6q/vold.fstab
deleted file mode 100644
index 79a0a35..0000000
--- a/sabreauto_6q/vold.fstab
+++ /dev/null
@@ -1,35 +0,0 @@
-## Vold 2.0 Generic fstab
-## - San Mehat (san@android.com)
-##
-
-#######################
-## Regular device mount
-##
-## Format: dev_mount <label> <mount_point> <part> <sysfs_path1...>
-## label        - Label for the volume
-## mount_point  - Where the volume will be mounted
-## part         - Partition # (1 based), or 'auto' for first usable partition.
-## <sysfs_path> - List of sysfs paths to source devices
-######################
-
-## Example of a standard sdcard mount for the emulator / Dream
-# Mounts the first usable partition of the specified device
-#for internal emmc as primary storage
-#dev_mount sdcard /mnt/sdcard auto /devices/platform/mxsdhci.2/mmc_host/mmc0
-#for external sdcard as primary storage
-dev_mount extsd  /mnt/extsd auto /devices/platform/sdhci-esdhc-imx.0/mmc_host/mmc1/
-#dev_mount extsd /mnt/sdcard/extsd auto /devices/platform/mxsdhci.0/mmc_host/mmc1
-#used for usb host
-#dev_mount udisk /mnt/udisk auto /devices/platform/fsl-ehci.0/usb1
-#used for usb otg host
-#dev_mount udisk /mnt/udisk auto /devices/platform/fsl-ehci.0/usb
-#used for all usb host
-dev_mount udisk /mnt/udisk auto /devices/platform/fsl-ehci
-
-
-## Example of a dual card setup
-# dev_mount left_sdcard  /sdcard1  auto /devices/platform/goldfish_mmc.0 /devices/platform/msm_sdcc.2/mmc_host/mmc1
-# dev_mount right_sdcard /sdcard2  auto /devices/platform/goldfish_mmc.1 /devices/platform/msm_sdcc.3/mmc_host/mmc1
-
-## Example of specifying a specific partition for mounts
-# dev_mount sdcard /sdcard 2 /devices/platform/goldfish_mmc.0 /devices/platform/msm_sdcc.2/mmc_host/mmc1
diff --git a/sabresd_6dq/BoardConfig.mk b/sabresd_6dq/BoardConfig.mk
index 080aa9f..447b596 100755
--- a/sabresd_6dq/BoardConfig.mk
+++ b/sabresd_6dq/BoardConfig.mk
@@ -10,6 +10,16 @@ include device/fsl-proprietary/gpu-viv/fsl-gpu.mk
 BUILD_TARGET_FS ?= ext4
 include device/fsl/imx6/imx6_target_fs.mk
 
+ifeq ($(BUILD_TARGET_FS),ubifs)
+# build ubifs for nand devices
+PRODUCT_COPY_FILES +=	\
+	device/fsl/sabresd_6dq/fstab_nand.freescale:root/fstab.freescale
+else
+# build for ext4
+PRODUCT_COPY_FILES +=	\
+	device/fsl/sabresd_6dq/fstab.freescale:root/fstab.freescale
+endif # BUILD_TARGET_FS
+
 
 TARGET_BOOTLOADER_BOARD_NAME := SABRESD
 PRODUCT_MODEL := SABRESD-MX6DQ
diff --git a/sabresd_6dq/fstab.freescale b/sabresd_6dq/fstab.freescale
new file mode 100644
index 0000000..dcbcd14
--- /dev/null
+++ b/sabresd_6dq/fstab.freescale
@@ -0,0 +1,12 @@
+# Android fstab file.
+#<src>                                                  <mnt_point>         <type>    <mnt_flags and options>                       <fs_mgr_flags>
+# The filesystem that contains the filesystem checker binary (typically /system) cannot
+# specify MF_CHECK, and must come before any filesystems that do specify MF_CHECK
+
+/devices/platform/sdhci-esdhc-imx.2/mmc_host/mmc1 /mnt/extsd vfat defaults voldmanaged=sdcard:auto
+/devices/platform/fsl-ehci /mnt/udisk vfat defaults voldmanaged=sdcard:auto
+/dev/block/mmcblk0p5    /system  ext4    ro                                                                               wait
+/dev/block/mmcblk0p4    /data    ext4    nosuid,nodev,nodiratime,noatime,nomblk_io_submit,noauto_da_alloc,errors=panic    wait,encryptable=footer
+/dev/block/mmcblk0p6    /cache   ext4    nosuid,nodev,nomblk_io_submit				                          wait
+/dev/block/mmcblk0p7    /device  ext4    ro,nosuid,nodev				                                  wait
+
diff --git a/sabresd_6dq/vold.fstab b/sabresd_6dq/vold.fstab
deleted file mode 100644
index db230c5..0000000
--- a/sabresd_6dq/vold.fstab
+++ /dev/null
@@ -1,38 +0,0 @@
-## Vold 2.0 Generic fstab
-## - San Mehat (san@android.com)
-##
-
-#######################
-## Regular device mount
-##
-## Format: dev_mount <label> <mount_point> <part> <sysfs_path1...>
-## label        - Label for the volume
-## mount_point  - Where the volume will be mounted
-## part         - Partition # (1 based), or 'auto' for first usable partition.
-## <sysfs_path> - List of sysfs paths to source devices
-######################
-
-## Example of a standard sdcard mount for the emulator / Dream
-# Mounts the first usable partition of the specified device
-#for internal emmc as primary storage
-#dev_mount sdcard /mnt/sdcard auto /devices/platform/mxsdhci.2/mmc_host/mmc0
-#for external sdcard as primary storage
-#dev_mount sdcard /mnt/sdcard auto /devices/platform/sdhci-esdhc-imx.3/mmc_host/mmc0/
-#dev_mount extsd /mnt/extsd auto /devices/platform/mxsdhci.0/mmc_host/mmc1
-#used for usb host
-#dev_mount udisk /mnt/udisk auto /devices/platform/fsl-ehci.0/usb1
-#used for usb otg host
-#dev_mount udisk /mnt/udisk auto /devices/platform/fsl-ehci.0/usb
-#used for all usb host
-dev_mount udisk /mnt/udisk auto /devices/platform/fsl-ehci
-#mount SDHC4 SD card /mnt/sdcard as primary storage forMX6Q SABER_LITE RevC
-dev_mount extsd /mnt/extsd auto /devices/platform/sdhci-esdhc-imx.2/mmc_host/mmc1
-#mount SDHC3 TF card to /mnt/extsd as external storage forMX6Q SABER_LITE RevC
-#dev_mount sdcard /mnt/sdcard 4 /devices/platform/sdhci-esdhc-imx.3/mmc_host/mmc0
-
-## Example of a dual card setup
-# dev_mount left_sdcard  /sdcard1  auto /devices/platform/goldfish_mmc.0 /devices/platform/msm_sdcc.2/mmc_host/mmc1
-# dev_mount right_sdcard /sdcard2  auto /devices/platform/goldfish_mmc.1 /devices/platform/msm_sdcc.3/mmc_host/mmc1
-
-## Example of specifying a specific partition for mounts
-# dev_mount sdcard /sdcard 2 /devices/platform/goldfish_mmc.0 /devices/platform/msm_sdcc.2/mmc_host/mmc1
-- 
1.8.0

